website/content/docs/commands/validate.mdx

```diff
@@ -39,6 +45,15 @@ Errors validating build 'vmware'. 1 error(s) occurred:
   source block's "name" label, unless an in-build source definition adds the
   "name" configuration option.
 
+- `-no-warn-on-undeclared-var` - Silence warnings when the template assigns
+  undeclared variables. This is especially relevant for HCL2 templates with
+  .pkrvars.hcl dependencies, as defining a value for a variable is not enough on
+  its own for Packer to function, there also needs to be some variable definitions
+  in the template. However, for some cases, templates may only declare what they
+  locally need for building, and reference a global variable file with more variable
+  assignations than there are definitions. This would yield a warning for each
+  assigned and non-declared variable. This option will disable those warnings.
+
```

Comment:
```suggestion
- `-no-warn-on-undeclared-var` - Silence warnings when the a variable definition file contains variable assignments for undeclared variables. This can occur when using a var-file that contains a large amount of unused variables for a given HCL2 template. For HCL2 template defining a value for a variable in a var-file is not enough on its own for Packer to function, as there also needs to be a variable block definition in the template files `pkr.hcl` for the variable. By default `packer validate` will warn when a var-file contains one or more undeclared variables. 
```